### ISSUE TITLE
`zeromq` package is missing optional features provided by other distributions

### ISSUE THREAD

**Issue by PJungkamp** [NixOS/nixpkgs#337508]
### Describe the bug

The `zeromq` package providing `libzmq` is missing several features that other distributions enable by default.

Some examples:
- Upstream Debian Packaging: https://github.com/zeromq/libzmq/blob/1f4dd54f94009cb57bb68bfc856edc1c0bd3df06/packaging/debian/rules#L46
- Arch Linux Packaging: https://gitlab.archlinux.org/archlinux/packaging/packages/zeromq/-/blob/main/PKGBUILD?ref_type=heads#L53-67
- Fedora Packaging: https://src.fedoraproject.org/rpms/zeromq/blob/rawhide/f/zeromq.spec#_70

Features like `ENABLE_CURVE` and `WITH_LIBSODIUM` should probably be enabled by default. I was expecting curve support from the package and even saw `libsodium` in the package's inputs, yet I ran into an `ENOTSUP` on `zmq_curve_keypair`.
- See the unused `libsodium` input: https://github.com/NixOS/nixpkgs/blob/31fb372aa93926f3e05401c7474cc560e197c765/pkgs/development/libraries/zeromq/4.x.nix#L7
- See the ENOTSUP error here: https://libzmq.readthedocs.io/en/latest/zmq_curve_keypair.html

### Expected behavior

An we should probably provide build options similar to the upstream RHEL packaging. See https://github.com/zeromq/libzmq/blob/1f4dd54f94009cb57bb68bfc856edc1c0bd3df06/packaging/redhat/zeromq.spec#L184-L193

Most should probably be enabled by default.

### Notify maintainers

@fpletz 

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by smancill** [comment]
I'll work on this tomorrow, but I'd rather follow what [Debian does](https://packages.debian.org/trixie/libzmq5):

```
override_dh_auto_configure:
	dh_auto_configure -- --with-pgm --with-libsodium --with-norm=yes \
		--with-libgssapi_krb5=yes --disable-Werror
```

Certainly we need to enable the flag to use `libsodium`.

**Comment by fpletz** [comment]
> About CurveZMQ, I'd say the best option is to provide an input flag as false that can be overridden by those who prefer it.

Since we depend on `libsodium` anyway I don't think there is any harm in enabling CurveZMQ by default - even without a flag. Are there negative side effects when enabling CurveZMQ?

**Comment by PJungkamp** [comment]
> ```
> override_dh_auto_configure:
> 	dh_auto_configure -- --with-pgm --with-libsodium --with-norm=yes \
> 		--with-libgssapi_krb5=yes --disable-Werror
> ```

Is OpenPGM or NORM even present in `nixpkgs`? I don't think that we can enable these.

**Comment by smancill** [comment]
> > About CurveZMQ, I'd say the best option is to provide an input flag as false that can be overridden by those who prefer it.
> 
> Since we depend on `libsodium` anyway I don't think there is any harm in enabling CurveZMQ by default - even without a flag. Are there negative side effects when enabling CurveZMQ?

Sorry, I was confusing the CurveCP protocol, which is supported by enabling `libsodium`,
with CurveZMQ, which is a CurveCP implementation but its a whole different library that depends on `libzmq`.

There is no flag related to CurveZMQ on ZeroMQ. But we do need to enable the flag to build with CurveCP support.


> > ```
> > override_dh_auto_configure:
> > 	dh_auto_configure -- --with-pgm --with-libsodium --with-norm=yes \
> > 		--with-libgssapi_krb5=yes --disable-Werror
> > ```
> 
> Is OpenPGM or NORM even present in `nixpkgs`? I don't think that we can enable these.

Yes, sure, if the flag doesn't apply to Nix then it will not be considered.

**Comment by smancill** [comment]
Closed by #338064.

(Why it wasn't closed automatically if it was linked?)
